protocol/messenger.go

```diff
@@ -2812,6 +2822,20 @@ func (m *Messenger) ValidateTransactions(ctx context.Context, addresses []types.
 	return &response, nil
 }
 
+func (m *Messenger) addResponses(messages []*Message) ([]*Message, error) {
```

Comment:
What this method does is actually pulling messages not adding them to the messenger/store as a name suggests. Should we rename it? 

Or it can be a separate function:
```go
messageAndResponseIDs := pickMessageAndResponseIDs(messageState.Response.Messages)
messagesWithResponses, err := m.persistence.MessagesByIDs(messageAndResponseIDs)
```